packages/apollo/src/commands/service/list.ts

```diff
@@ -0,0 +1,173 @@
+  graphName: string | undefined;
+  frontendUrl: string | undefined;
+}): string {
+  let result = "";
+  if (
+    !implementingServices ||
+    implementingServices.__typename === "NonFederatedImplementingService"
+  ) {
+    result =
+      "\nThis graph is not federated, there are no services composing the graph";
+  } else if (implementingServices.services.length === 0) {
+    result = "\nThere are no services on this federated graph";
+  } else {
+    // Create a sorted list of the services.
+    const sortedImplementingServices = sortBy<
+      ListServices_service_implementingServices_FederatedImplementingServices_services
+    >(implementingServices.services, [service => service.name.toUpperCase()]);
+
+    table(
+      [...sortedImplementingServices.map(formatImplementingService)].filter(
+        Boolean
+      ),
+      {
+        columns: [
+          { key: "name", label: "Name" },
+          { key: "url", label: "Url" },
+          { key: "updatedAt", label: "Last Updated At" }
+        ],
+        // Override `printHeader` so we don't print a header
+        printHeader: () => {},
+        // The default `printLine` will output to the console; we want to capture the output so we can test
+        // it.
+        printLine: line => {
+          result += `\n${line}`;
+        }
+      }
+    );
+    const serviceListUrlEnding = `/service/${graphName}/service-list`;
+    const targetUrl = frontendUrl
+      ? `${frontendUrl}${serviceListUrlEnding}`
+      : `https://engine.apollographql.com${serviceListUrlEnding}`;
+    result += `\n\nView full details at: ${targetUrl}`;
+  }
+  return result;
+}
+
+export default class ServiceList extends ProjectCommand {
+  static description = "List the services in a graph";
+  static flags = {
+    ...ProjectCommand.flags,
+    tag: flags.string({
+      char: "t",
+      description: "The published tag to list the services from"
+    })
+    // json: flags.boolean({
+    //   description:
+    //     "Output result in json, which can then be parsed by CLI tools such as jq.",
+    //   exclusive: ["markdown"]
+    // }),
+  };
+
+  async run() {
+    // @ts-ignore we're going to populate `taskOutput` later
+    const taskOutput: TasksOutput = {};
+
+    let schema: GraphQLSchema | undefined;
+    try {
+      await this.runTasks<TasksOutput>(({ config, flags, project }) => {
+        if (!isServiceProject(project)) {
+          throw new Error(
+            "This project needs to be configured as a service project but is configured as a client project. Please see bit.ly/2ByILPj for help regarding configuration."
+          );
+        }
+
+        /**
+         * Name of the graph being checked. `engine` is an example of a graph.
+         *
+         * A graph can be either a monolithic schema or the result of composition a federated schema.
+         */
+        const graphName = config.name;
+        const tag = flags.tag || "current"; //|| config.tag || "current";
```

Comment:
forgot to continue investigating this - looking today

Reply:
I don't actually have a good answer for how to capture without copy/pasting. If the copy/paste works, then do that I guess 🤷‍♂

Reply:
This may be causing tests to fail ;(